action.yaml

```diff
@@ -50,35 +67,35 @@ runs:
           echo "EOF" >> $GITHUB_ENV
         fi
 
-    - name: "Check for new changes in dependencies"
-      uses: brettcannon/check-for-changed-files@v1
-      with:
-        file-pattern: ${{ env.FILE_PATTERN }}
-        failure-message: "No changes in ${ file-pattern }"
-
     - name: Clone Packj repo
+      if: env.CHECK_PASSED == 'true'
       shell: bash
       run: git clone https://github.com/ossillate-inc/packj
 
     - name: Setup python & pip
+      if: env.CHECK_PASSED == 'true'
       uses: actions/setup-python@v4
       with:
         python-version: "3.10"
         cache: pip
 
     - name: Create virtual environment
+      if: env.CHECK_PASSED == 'true'
       shell: bash
       run: python3 -m venv venv
 
     - name: Activate virtual environment
+      if: env.CHECK_PASSED == 'true'
       shell: bash
       run: source venv/bin/activate
 
     - name: Install Packj's dependencies
+      if: env.CHECK_PASSED == 'true'
       shell: bash
       run: pip3 install -r ./packj/requirements.txt
 
     - name: Auditing deps with Packj
+      if: env.CHECK_PASSED == 'true'
       shell: bash
       run: |
         input=$(echo "${{ inputs.DEPENDENCY_FILES }}" | sed 's/,/ /g') 
```

Comment:
we are parsing deps file multiple times: first at line 45 name: Parse dependency files then again at line 101. We could store the output in ENV and reuse here.